### DISCUSSION TITLE
Truenas Scale Docker (OS Error 2)

### DISCUSSION THREAD

**Discussion by jsdukeboy08** [jlesage/docker-czkawka#39] [Q&A] [answered]
I have this installed via docker on truenas scale 24.10, I am able to scan and see the files in my two folders that I have added to the container, but when I try to hardlink the 2 files I get an error "Failed to hardlink /storage/test2/filename, reason No such file or directory (os error 2).  Both files are located in the same dataset.

**Comment by jlesage** [comment]
> Both files are located in the same dataset.

Are there also in the same Docker volume/mount ? What is the path of both files inside the container ?

**Reply by jsdukeboy08** [reply]
This is how I have them set currently, I have tried mounting them in their own directory, without the storage directory, as well as just both of them to the same /storage directory without test/test2, but then it didn't find the duplicates.  Does the config directory also need to be in the same dataset?  I have my Truenas APPS set to run from a different NVMe pool and that is where the configs are stored
- /mnt/SPEED/docker/czkawka:/config:rw
- /mnt/BFD/Media/test:/storage/test:rw
- /mnt/BFD/Media/test2:/storage/test2:rw

**Reply by jlesage** [reply, marked as answer]
This setup won't allow hard-linking. What should be done is `/mnt/BFD/Media:/storage:rw`. This should not prevent Czkawka to find duplicates. Did you check that all the files and folders under `Media` have the save ownership and permissions ?

> Does the config directory also need to be in the same dataset?

No, it is not required.

**Reply by jsdukeboy08** [reply]
Well dang, I was hosed by my own testing method, the actual path was going to be /mnt/BFD/Media:/storage:rw but I wanted to test it before letting it loose to scan my entire dataset.  Everything is working now that I set it to your recommendation.  Thank you very much for the quick reply and for maintaining this!

**Reply by jlesage** [reply]
Great!
